simulator/func_sim/syscall/mars_calls.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * mars_calls.h - implementation of MARS syscalls
+ * @author Vyacheslav Kompan kompan.vo@phystech.edu
+ * Copyright 2018 MIPT-MIPS
+ */
+
+#ifndef MARS_CALLS_H
+#define MARS_CALLS_H
+
+/* Simulator modules */
+#include "func_sim/rf/rf.h"
+#include <infra/exception.h>
+#include "infra/types.h"
+/* Generic C & C++ */
+#include <cassert>
+#include <map>
+#include <memory>
+#include <iostream>
+
+
+struct TerminatedByExit final : Exception {
+    TerminatedByExit() : Exception( "Terminated", "exit called") {}
+};
+
+template <typename ISA>
```

Comment:
That should go to a dedicated header file

Reply:
Should it be placed in `infra/`?

Reply:
If we continue in 'Simulator - Memory - Syscall' same-rank model, then maybe the best decision is to put everyhing to a separate folder, but I'm ok with `infra` as well.